streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -0,0 +1,557 @@
+            }
+        },
+        IN_MEMORY_KV {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.inMemoryKeyValueStore(STORE_NAME);
+            }
+        },
+        IN_MEMORY_LRU {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.lruMap(STORE_NAME, 100);
+            }
+        },
+        ROCKS_KV {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentKeyValueStore(STORE_NAME);
+            }
+        },
+        TIME_ROCKS_KV {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentTimestampedKeyValueStore(STORE_NAME);
+            }
+        },
+        IN_MEMORY_WINDOW {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.inMemoryWindowStore(STORE_NAME, Duration.ofDays(1), WINDOW_SIZE,
+                    false);
+            }
+        },
+        ROCKS_WINDOW {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentWindowStore(STORE_NAME, Duration.ofDays(1), WINDOW_SIZE,
+                    false);
+            }
+        },
+        TIME_ROCKS_WINDOW {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentTimestampedWindowStore(STORE_NAME, Duration.ofDays(1),
+                    WINDOW_SIZE, false);
+            }
+        },
+        IN_MEMORY_SESSION {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.inMemorySessionStore(STORE_NAME, Duration.ofDays(1));
+            }
+        },
+        ROCKS_SESSION {
+            @Override
+            public StoreSupplier<?> supplier() {
+                return Stores.persistentSessionStore(STORE_NAME, Duration.ofDays(1));
+            }
+        };
+
+        public abstract StoreSupplier<?> supplier();
+
+        public boolean global() {
+            return false;
+        }
+    }
+
+    @Parameterized.Parameters(name = "cache={0}, log={1}, supplier={2}")
+    public static Collection<Object[]> data() {
+        final List<Object[]> values = new ArrayList<>();
+        for (final boolean cacheEnabled : Arrays.asList(true, false)) {
+            for (final boolean logEnabled : Arrays.asList(true, false)) {
+                for (final StoresToTest toTest : StoresToTest.values()) {
+                    values.add(new Object[]{cacheEnabled, logEnabled, toTest.name()});
+                }
+            }
+        }
+        return values;
+    }
+
+    public IQv2StoreIntegrationTest(
+        final boolean cache,
+        final boolean log,
+        final String storeToTest) {
+
+        this.cache = cache;
+        this.log = log;
+        this.storeToTest = StoresToTest.valueOf(storeToTest);
+    }
+
+    @BeforeClass
```

Comment:
To save time, we start Kafka and create and populate our input topic just once before all tests. None of the tests require writing new data, so this works.